src/Service/CommercialLicense.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Shopware\ServiceBundle\Service;
+
+use Lcobucci\Clock\SystemClock;
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Encoding\JoseEncoder;
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Lcobucci\JWT\Signer\Rsa\Sha512;
+use Lcobucci\JWT\Token\Parser;
+use Lcobucci\JWT\UnencryptedToken;
+use Lcobucci\JWT\Validation\Constraint\PermittedFor;
+use Lcobucci\JWT\Validation\Constraint\SignedWith;
+use Lcobucci\JWT\Validation\Constraint\StrictValidAt;
+use Lcobucci\JWT\Validation\RequiredConstraintsViolated;
+use RuntimeException;
+use Shopware\App\SDK\Shop\ShopRepositoryInterface;
+use Shopware\ServiceBundle\Entity\Shop;
+
+class CommercialLicense
+{
+    /**
+     * @param ShopRepositoryInterface<Shop> $shopRepository
+     */
+    public function __construct(
+        private readonly ShopRepositoryInterface $shopRepository,
+    ) {}
+
+    /**
+     * @throws RuntimeException
+     *
+     * @return array<string, string>
+     */
+    public function getInfo(string $shopId): array
```

Comment:
My suggestion is to create a new object, like `LicenseInfo`, to encapsulate information such as `license_domain`, `issued_at`, `expired_at`, etc., to maintain consistency throughout the project. And return null if there is no license or an invalid license.

The current return array is fine, but we always need to check if an `error` exists in the return. Additionally, if the return array's information is updated, we might end up with outdated data when calling this function from other services (apps).

Reply:
DTO > array :wink: